src/main/java/gui/EditRulePopup.java

```diff
@@ -0,0 +1,285 @@
+	 */
+	private JPanel createNamePanel() {
+		namePanel = new JPanel();
+		namePanel.setLayout(new GridLayout(1, 2));
+		JLabel nameLabel = new JLabel("Name: ", SwingConstants.LEFT);
+		nameText = new JTextField();
+		namePanel.add(nameLabel);
+		namePanel.add(nameText);
+		return namePanel;
+	}
+
+	/**
+	 * 
+	 * Creates and returns the JPanel responsible for holding both the current list
+	 * of rule metrics, and the line which allows the addition of more metrics.
+	 * 
+	 * @return Returns the JPanel responsible for holding both the current list
+	 * of rule metrics, and the line which allows the addition of more metrics.
+	 */
+	private JPanel createMetricsPanel() {
+		metricsListPanel = new JPanel();
+		metricsListPanel.setLayout(new BoxLayout(metricsListPanel, BoxLayout.Y_AXIS));
+		metricsScrollpane = new JScrollPane(metricsListPanel, JScrollPane.VERTICAL_SCROLLBAR_ALWAYS,
+				JScrollPane.HORIZONTAL_SCROLLBAR_NEVER);
+		fillMetricsListPanel();
+
+		addNewMetricPanel = new JPanel();
+		createAddMetricPanel();
+
+		metricsPanel = new JPanel();
+		metricsPanel.setLayout(new BorderLayout());
+		metricsPanel.add(addNewMetricPanel, BorderLayout.NORTH);
+		metricsPanel.add(metricsScrollpane, BorderLayout.CENTER);
+		return metricsPanel;
+	}
+
+	/**
+	 * Creates and returns the JPanel responsible for holding the JButtons which
+	 * allow all metrics to be cleared, or the rule to be saved.
+	 * 
+	 * @return Returns the JPanel responsible for holding the JButtons which
+	 * allow all metrics to be cleared, or the rule to be saved.
+	 */
+	private JPanel createControlPanel() {
+		controlPanel = new JPanel();
+		controlPanel.setLayout(new GridLayout(1, 3));
+
+		JButton clearButton = new JButton("Clear Metrics");
+		JButton deleteButton = new JButton("Delete Rule");
+		JButton saveButton = new JButton("Save Rule");
+
+		clearButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				clearMetricsListPanel();
+			}
+		});
+
+		deleteButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				if (nameText.getText().isEmpty()) {
+					JOptionPane.showMessageDialog(null, "Please insert a rule name!");
+				} else {
+					// Add functionality here.
+				}
+			}
+		});
+
+		saveButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				if (nameText.getText().isEmpty()) {
+					JOptionPane.showMessageDialog(null, "Please insert a rule name!");
+				} else {
+					// Add functionality here.
+				}
+			}
+		});
+
+		controlPanel.add(clearButton);
+		controlPanel.add(deleteButton);
+		controlPanel.add(saveButton);
+		return controlPanel;
+
+	}
+
+	/**
+	 * This method both returns and creates the JPanel holding the line which allows
+	 * users to add new metrics into the metrics list. TODO: The currently
+	 * hard-coded values of the JBoxes are to be replaced with the correct, final
+	 * values once the ENUMs are created.

```

Comment:
Keep the TODO on its own line, otherwise it gets lost.
Also same as previous comments - no use in repeating the exact same stuff